kbase-extension/static/kbase/js/widgets/appWidgets2/validation.js

```diff
@@ -632,13 +697,28 @@ define([
             };
         }
 
+        /**
+         * Basically casts undefined -> null, otherwise returns the given string.
+         * @param {String} value the value to verify
+         * @returns
+         */
         function importTextString(value) {
             if (value === undefined || value === null) {
                 return null;
             }
             return value;
         }
 
+        /**
+         * This imports an integer string under the following restrictions.
+         * 1. Returns null if the value is undefined, null, or a whitespace-only string.
+         * 2. Throws an error if the value is not a string
+         * 3. Attempts to cast the string to an integer. If it fails, throws an Error.
+         * 4. If nonIntError is given and an error occurs, then nonIntError is thrown.
+         * @param {String} value expected to be an integer string, will be cast into an int if so.
+         * @param {String} nonIntError
+         * @returns
+         */
```

Comment:
Also TODO: merge `importIntString` and `importFloatString` as they're nearly identical.